From 1ba18945d2f7e9bd5c8aa3b02f0ceccf64059276 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 12 Feb 2021 15:58:49 +0100
Subject: [PATCH 4/4] Revert "radeon, amdgpu: Firmware is required for DRM and
 KMS on R600 onward"

Although this patch helps in Debian, not all distributions using Zen
Kernrel have the same folder structure (/lib/firmware).  And as @Atemu
pointed out in #200, the patch is technically incredibly old for kernel
standards so it may only do harm than good.  Revert this patch.

This reverts commit 64ce4b2a1cbca4717e5606612b9166178361bcd6.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 29 ------------------------
 drivers/gpu/drm/radeon/radeon_drv.c     | 30 -------------------------
 2 files changed, 59 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index 5f6496926..7169fb5e3 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -36,8 +36,6 @@
 #include <linux/vga_switcheroo.h>
 #include <drm/drm_probe_helper.h>
 #include <linux/mmu_notifier.h>
-#include <linux/namei.h>
-#include <linux/path.h>
 
 #include "amdgpu.h"
 #include "amdgpu_irq.h"
@@ -1124,28 +1122,6 @@ MODULE_DEVICE_TABLE(pci, pciidlist);
 
 static const struct drm_driver amdgpu_kms_driver;
 
-/* Test that /lib/firmware/amdgpu is a directory (or symlink to a
- * directory).  We could try to match the udev search path, but let's
- * keep it simple.
- */
-static bool amdgpu_firmware_installed(void)
-{
-#if IS_BUILTIN(CONFIG_DRM_AMDGPU)
-	/* It may be too early to tell.  Assume it's there. */
-	return true;
-#else
-	struct path path;
-
-	if (kern_path("/lib/firmware/amdgpu", LOOKUP_DIRECTORY | LOOKUP_FOLLOW,
-		      &path) == 0) {
-		path_put(&path);
-		return true;
-	}
-
-	return false;
-#endif
-}
-
 static int amdgpu_pci_probe(struct pci_dev *pdev,
 			    const struct pci_device_id *ent)
 {
@@ -1210,11 +1186,6 @@ static int amdgpu_pci_probe(struct pci_dev *pdev,
 	}
 #endif
 
-	if (!amdgpu_firmware_installed()) {
-		DRM_ERROR("amdgpu requires firmware installed\n");
-		return -ENODEV;
-	}
-
 	/* Get rid of things like offb */
 	ret = drm_fb_helper_remove_conflicting_pci_framebuffers(pdev, "amdgpudrmfb");
 	if (ret)
diff --git a/drivers/gpu/drm/radeon/radeon_drv.c b/drivers/gpu/drm/radeon/radeon_drv.c
index 80fccfe8d..e45d7344a 100644
--- a/drivers/gpu/drm/radeon/radeon_drv.c
+++ b/drivers/gpu/drm/radeon/radeon_drv.c
@@ -49,8 +49,6 @@
 #include <drm/drm_probe_helper.h>
 #include <drm/drm_vblank.h>
 #include <drm/radeon_drm.h>
-#include <linux/namei.h>
-#include <linux/path.h>
 
 #include "radeon_drv.h"
 #include "radeon.h"
@@ -291,28 +289,6 @@ MODULE_DEVICE_TABLE(pci, pciidlist);
 
 static const struct drm_driver kms_driver;
 
-/* Test that /lib/firmware/radeon is a directory (or symlink to a
- * directory).  We could try to match the udev search path, but let's
- * keep it simple.
- */
-static bool radeon_firmware_installed(void)
-{
-#if IS_BUILTIN(CONFIG_DRM_RADEON)
-	/* It may be too early to tell.  Assume it's there. */
-	return true;
-#else
-	struct path path;
-
-	if (kern_path("/lib/firmware/radeon", LOOKUP_DIRECTORY | LOOKUP_FOLLOW,
-		      &path) == 0) {
-		path_put(&path);
-		return true;
-	}
-
-	return false;
-#endif
-}
-
 static int radeon_pci_probe(struct pci_dev *pdev,
 			    const struct pci_device_id *ent)
 {
@@ -353,12 +329,6 @@ static int radeon_pci_probe(struct pci_dev *pdev,
 	if (vga_switcheroo_client_probe_defer(pdev))
 		return -EPROBE_DEFER;
 
-	if ((ent->driver_data & RADEON_FAMILY_MASK) >= CHIP_R600 &&
-	    !radeon_firmware_installed()) {
-		DRM_ERROR("radeon kernel modesetting for R600 or later requires firmware installed\n");
-		return -ENODEV;
-	}
-
 	/* Get rid of things like offb */
 	ret = drm_fb_helper_remove_conflicting_pci_framebuffers(pdev, "radeondrmfb");
 	if (ret)
-- 
2.30.0.452.gfb7fa4a1fd

